lib/ReactViews/DataCatalog/DataCatalogReference.tsx

```diff
@@ -0,0 +1,104 @@
+import CatalogMemberMixin from "../../ModelMixins/CatalogMemberMixin";
+import ReferenceMixin from "../../ModelMixins/ReferenceMixin";
+import CommonStrata from "../../Models/Definition/CommonStrata";
+import Model from "../../Models/Definition/Model";
+import Terria from "../../Models/Terria";
+import ViewState from "../../ReactViewModels/ViewState";
+import CatalogMemberReferenceTraits from "../../Traits/TraitsClasses/CatalogMemberReferenceTraits";
+import CatalogGroup from "./CatalogGroup";
+import CatalogItem, { ButtonState } from "./CatalogItem";
+import toggleItemOnMapFromCatalog, {
+  Op as ToggleOnMapOp
+} from "./toggleItemOnMapFromCatalog";
+
+interface Props {
+  reference: ReferenceMixin.Instance &
+    CatalogMemberMixin.Instance &
+    Model<CatalogMemberReferenceTraits>;
+  viewState: ViewState;
+  terria: Terria;
+  onActionButtonClicked?: (item: Props["reference"]) => void;
+  isTopLevel: boolean;
+}
+
+export default observer(function DataCatalogReference({
+  reference,
+  viewState,
+  terria,
+  onActionButtonClicked,
+  isTopLevel
+}: Props) {
+  const setPreviewedItem = () =>
+    viewState.viewCatalogMember(reference, true, CommonStrata.user);
+
+  const add = async (event: React.MouseEvent<HTMLButtonElement>) => {
+    const keepCatalogOpen = event.shiftKey || event.ctrlKey;
+
+    if (onActionButtonClicked) {
+      onActionButtonClicked(reference);
+      return;
+    }
+
+    if (defined(viewState.storyShown)) {
+      runInAction(() => {
+        viewState.storyShown = false;
+      });
+    }
+
+    if (
+      defined((reference as any).invoke) ||
+      viewState.useSmallScreenInterface
+    ) {
+      await setPreviewedItem();
+    } else {
+      await toggleItemOnMapFromCatalog(viewState, reference, keepCatalogOpen, {
+        [ToggleOnMapOp.Add]: DataSourceAction.addFromCatalogue,
+        [ToggleOnMapOp.Remove]: DataSourceAction.removeFromCatalogue
+      });
+    }
+  };
+
+  const isSelected = addedByUser(reference)
+    ? viewState.userDataPreviewedItem === reference
+    : viewState.previewedItem === reference;
+
+  const path = getPath(reference, " -> ");
+
+  return (
+    <>
+      {reference.isGroup ? (
+        <CatalogGroup
+          text={reference.name || "..."}
+          isPrivate={reference.isPrivate}
+          title={path}
+          onClick={setPreviewedItem}
+          topLevel={isTopLevel}
+          loading={reference.isLoadingReference}
+          open={reference.isLoadingReference}
+        />
+      ) : null}
+      <CatalogItem
+        onTextClick={setPreviewedItem}
+        selected={isSelected}
+        text={reference.name || "..."}
+        isPrivate={reference.isPrivate}
+        title={path}
+        btnState={
+          reference.isLoadingReference
+            ? ButtonState.Loading
+            : reference.isFunction
+            ? ButtonState.Stats
+            : ButtonState.Add
+        }
+        onBtnClick={reference.isFunction ? setPreviewedItem : add}
+      />
```

Comment:
`btnState` is missing the `viewState.useSmallScreenInterface` check (in `DataCatalogItem.tsx`) - so on small/mobile screen interfaces we are seeing "(+)" (add button) for references, which change to ">" (set preview button) after loading

<img width="494" alt="image" src="https://user-images.githubusercontent.com/6187649/144228349-948f2afd-0986-497b-bea0-f6e8ec7a665f.png">

http://ci.terria.io/fix-add-to-map/#configUrl=https://nationalmap.gov.au/config.json&share=s-jnii2PJAvbbSxv93GtO1GHKfJhV

Reply:
Alternatively, I'm not sure if `DataCatalogItem` and `DataCatalogReference` need to be separate files, they have very similar code - but that can be left for another day when we tsxify `DataCatalogMember.jsx`, `DataCatalogGroup.jsx`, `CatalogGroup.jsx`...

I nearly think `DataCatalog*` could be merged into a single `DataCatalogMember` file - but it might get a bit crowded